Celeste.Mod.mm/Mod/Everest/Everest.Loader.cs

```diff
@@ -484,12 +484,17 @@ public static class Loader {
                 Assembly asm = null;
                 if (!string.IsNullOrEmpty(meta.PathArchive)) {
                     bool returnEarly = false;
+                    string dllDirectory = Path.GetDirectoryName(meta.DLL);
                     using (ZipFile zip = new ZipFile(meta.PathArchive)) {
                         foreach (ZipEntry entry in zip.Entries) {
                             string entryName = entry.FileName.Replace('\\', '/');
-                            if (entryName == meta.DLL) {
-                                using (MemoryStream stream = entry.ExtractStream())
+
+                            if (Path.GetExtension(entryName) == ".dll") {
+                                using MemoryStream stream = entry.ExtractStream();
+                                if (entryName == meta.DLL)
                                     asm = Relinker.GetRelinkedAssembly(meta, Path.GetFileNameWithoutExtension(meta.DLL), stream);
+                                else if (entryName.StartsWith(dllDirectory)) // immediately load any other libraries to avoid reading zip files over and over again later
```

Comment:
```suggestion
                                else if (!string.IsNullOrEmpty(dllDirectory) && entryName.StartsWith(dllDirectory)) // immediately load any other libraries to avoid reading zip files over and over again later
```

Reply:
I'll just do `dllDirectory = Path.GetDirectoryName(meta.DLL) ?? "";`, accomplishes the same thing

Reply:
no, if `dllDirectory` is empty string, then `entryName.StartsWith(dllDirectory)` will always return true

Reply:
no, if `dllDirectory` is empty string, then `entryName.StartsWith(dllDirectory)` will always return true

Reply:
Actually yeah, in that case my approach is better - if your mod dll is in the top folder of your mod (`dllDirectory = ""`, happens quite often), then libraries should also be loadable from the top folder, which that `!NullOrEmpty` check would prevent

Reply:
okay, but now there's another question, `entryName.StartsWith(dllDirectory)` would return true if  entryName in meta.DLL's subfolder

Reply:
I was using `StartsWith` to support subfolders, but library dlls are always in the same directory anyway so yeah that makes more sense.